VeganVirus/VeganVirus/FruitThrow.cpp

```diff
@@ -0,0 +1,100 @@
+#include "FruitThrow.h"
+#include <stdlib.h>
+#include <math.h>
+
+const std::vector<const wchar_t*> FruitThrow::fruitImages = {
+	L"fruit0.png",
+	L"fruit1.png",
+	L"fruit2.png",
+	L"fruit3.png",
+	L"fruit4.png",
+	L"fruit5.png",
+	L"fruit6.png",
+	L"fruit7.png",
+	L"fruit8.png",
+};
+
+FruitThrow::FruitThrow(Draw* draw, MouseManager& mouseManager) : _draw(draw), _mouseManager(mouseManager)
+{
+	// randomize fields
+	_r += rand() % RADIUS_RANGE;
+
+	const wchar_t* path = FruitThrow::fruitImages[rand() % FruitThrow::fruitImages.size()];
+	this->_image = Draw::resizedBitmap(path, _r * 2, _r * 2);
+
+	_screen = draw->getScreenSize();
+	_x = (float)rand() / RAND_MAX * (_screen.x + 400) - 200;
+	_y = rand() % 3 ? _screen.y + 100 : -100;
+	_vx = (_screen.x / 2 - _x) * ((float)rand() / RAND_MAX * 0.5 + 0.4);
+	if (_y < 0)
+		_vy = 0;
+	else
+		_vy = -sqrt(2. * _g * _screen.y * ((float)rand() / RAND_MAX * 0.55 + 0.6));
+}
+
+FruitThrow::~FruitThrow()
+{
+	delete this->_image;
+}
+
+void FruitThrow::update(double dt)
+{
+	_x += _vx * dt;
+	_y += _vy * dt;
+	_vy += _g * dt;
+
+	// check wall bounce
+	if (_x < (_r - BOUNCE_MARGIN_X) && _vx < 0 || _x > _screen.x - (_r - BOUNCE_MARGIN_X) && _vx > 0)
+		_vx *= -1;
+	if (_y < (_r - BOUNCE_MARGIN_Y) && _vy < 0)
+		_vy *= -1;
+}
+
+void FruitThrow::draw()
+{
+	this->_draw->drawImage(this->_image, _x - _r, _y - _r);
+}
+
+bool FruitThrow::checkMouseCollision(double x1, double y1)
+{
+	double x2 = _x, y2 = _y;
+	double Dx = x1 - x2, Dy = y1 - y2;
+	double Dsqr = Dx * Dx + Dy * Dy;
+	if (Dsqr <= _r * _r)
+	{
+		// there's a collision
+		if (_collisionLock)
+			return false;
+		_collisionLock = true;
+		return true;
+	}
+	// no collision
+	_collisionLock = false;
+	return false;
+}
+
+DPoint_t FruitThrow::mouseCollision(double x1, double y1, double vx1, double vy1)
+{
+	double x2 = _x, y2 = _y;
+	double Dx = x1 - x2, Dy = y1 - y2;
```

Comment:
I think the variable name should not start with caps lock. If you don't have any mathematical reason you should replace `Dx` with `dX`

Reply:
the capital D stands for a vector named D.
I think it's a lot clearer writing `Dx` for the x value of the vector, or `Dsqr` for the square of the vector, rather than `dx` (which is generally understood as the difference in x, delta x), `d_x`, `dsqr`, `dSqr` etc.
Do you still want me to change it?

Reply:
it is your choice. like I said if you took it to a count and you think this is the right way you can keep it